components/unitStats/weaponTable.tsx

```diff
@@ -0,0 +1,784 @@
+      break;
+    case "at_gun":
+      icon = "/icons/races/german/symbols/75mm_at_gun_ger.webp";
+      break;
+    case "mortar":
+      icon = "/icons/races/common/symbols/mortar.webp";
+      break;
+    case "artillery_medium":
+      icon = "/icons/unit_status/bw2/designated_artillery_overwatch_active.webp";
+      break;
+    case "cannon":
+      icon = "/icons/unit_status/bw2/anti_tank.webp";
+      break;
+    case "artillery_small":
+      icon = "/icons/unit_status/bw2/active_artillery.webp";
+      break;
+    case "artillery_mobile":
+      icon = "/icons/common/units/symbols/artillery.webp";
+      break;
+    case "pistol":
+      icon = "/icons/common/units/symbols/unit_british_officer_symbol.webp";
+      break;
+    default:
+      icon = "/icons/races/common/symbols/building_weapon_support.webp";
+      break;
+  }
+  return internalSlash(icon);
+};
+
+export const WeaponTable = ({ inputData }: inputProps) => {
+  tableData = inputData;
+
+  const [search, setSearch] = useState("");
+  const [factionFilter, setFactionFilter] = useState([] as string[]);
+  const [typeFilter, setTypeFilter] = useState([] as string[]);
+
+  const [updateFlag, updateTable] = useState(true);
+  const [sortedData, setSortedData] = useState(tableData);
+  const [sortBy, setSortBy] = useState<keyof WeaponTableRow | null>(null);
+  const [reverseSortDirection, setReverseSortDirection] = useState(false);
+
+  const [debouncedSearch] = useDebouncedValue(search, 600);
+
+  useEffect(() => {
+    setSortedData(
+      sortData(tableData, {
+        sortBy,
+        reversed: reverseSortDirection,
+        search: debouncedSearch,
+        factionFilter,
+        typeFilter,
+      }),
+    );
+  }, [debouncedSearch]);
+
+  function toggleFilter(filterValue: string, filterList: string[]) {
+    const filterValueIndex = filterList.indexOf(filterValue);
+    if (filterValueIndex < 0) filterList.push(filterValue);
+    else filterList.splice(filterValueIndex, 1);
+
+    setFactionFilter([...factionFilter]);
+    setTypeFilter([...typeFilter]);
+    setSortedData(
+      sortData(tableData, {
+        sortBy,
+        reversed: reverseSortDirection,
+        search: search,
+        factionFilter,
+        typeFilter,
+      }),
+    );
+  }
+
+  const setSorting = (field: keyof WeaponTableRow) => {
+    const reversed = field === sortBy ? !reverseSortDirection : false;
+    setReverseSortDirection(reversed);
+    setSortBy(field);
+    setSortedData(
+      sortData(tableData, { sortBy: field, reversed, search, factionFilter, typeFilter }),
+    );
+  };
+
+  const onTableLayoutChange = (colSetup: tableColSetup) => {
+    colSetup.visible = !colSetup.visible;
+    refresh();
+  };
+
+  const refresh = () => {
+    updateTable(!updateFlag);
+    return;
+  };
+
+  const rows = sortedData.map((unit) => mapRow(unit));
+
+  const cols = getTableHeader(sortBy as keyof WeaponTableRow, reverseSortDirection, setSorting);
+
+  return (
+    <ScrollArea>
```

Comment:
@KingDarBoja  I think that maybe we just need to wrap the table into this? So the table is scrollable but not the filters and seach